File: MarketPlace.Backend/MarketPlace/Microservices/Product Service/Services.Product.Domain/Models/Product.cs
namespace ProductService
{
    [Serializable]
    public class Product : Entity
    {
        public Guid ManufacturerId { get; set; }
        public string? Name { get; set; } 
        public string? Description { get; set; } 
        public string? Category { get; set; } 
        public string? Type { get; set; } 
        public List<Review> Reviews { get; set; } = [];
        public string? Image { get; set; } 
        public double Price { get; set; }
        public double Raiting { get; set; }
        public List<DateTime> ViewAt { get; set; } = [];

    }
}

File: MarketPlace.Frontend/market-place-client/src/app/data/interfaces/product.ts
import { Review } from "./review";

export interface Product {
    id: string;
    manufacturerId: string;
    name: string | null;
    description: string | null;
    category: string | null;
    type: string | null;
    reviews: Review[];
    image: string | null;
    price: number;
    raiting: number;
    viewAt: string[];
    creationDateTime: string;
  }
